extensions/cl_intel_device_attribute_query.asciidoc

```diff
@@ -0,0 +1,197 @@
+:CL_DEVICE_FEATURE_CAPABILITIES_INTEL: pass:q[`CL_DEVICE_<wbr>FEATURE_<wbr>CAPABILITIES_<wbr>INTEL`]
+endif::[]
+ifndef::backend-html5[]
+:cl_device_feature_capabilities_intel_TYPE: pass:q[`cl_device_&#8203;feature_&#8203;capabilities_&#8203;intel`]
+:CL_DEVICE_IP_VERSION_INTEL: pass:q[`CL_DEVICE_&#8203;IP_&#8203;VERSION_&#8203;INTEL`]
+:CL_DEVICE_ID_INTEL: pass:q[`CL_DEVICE_&#8203;ID_&#8203;INTEL`]
+:CL_DEVICE_NUM_SLICES_INTEL: pass:q[`CL_DEVICE_&#8203;NUM_&#8203;SLICES_&#8203;INTEL`]
+:CL_DEVICE_NUM_SUB_SLICES_PER_SLICE_INTEL: pass:q[`CL_DEVICE_&#8203;NUM_&#8203;SUB_&#8203;SLICES_&#8203;PER_&#8203;SLICE_&#8203;INTEL`]
+:CL_DEVICE_NUM_EUS_PER_SUB_SLICE_INTEL: pass:q[`CL_DEVICE_&#8203;NUM_&#8203;EUS_&#8203;PER_&#8203;SUB_&#8203;SLICE_&#8203;INTEL`]
+:CL_DEVICE_NUM_THREADS_PER_EU_INTEL: pass:q[`CL_DEVICE_&#8203;NUM_&#8203;THREADS_&#8203;PER_&#8203;EU_&#8203;INTEL`]
+:CL_DEVICE_FEATURE_CAPABILITIES_INTEL: pass:q[`CL_DEVICE_&#8203;FEATURE_&#8203;CAPABILITIES_&#8203;INTEL`]
+endif::[]
+
+= cl_intel_device_attribute_query
+
+== Name Strings
+
+`cl_intel_device_attribute_query`
+
+== Contact
+
+Ben Ashbaugh, Intel (ben 'dot' ashbaugh 'at' intel 'dot' com)
+
+== Contributors
+
+// spell-checker: disable
+Ben Ashbaugh, Intel +
+Maciej Dziuban, Intel +
+Rafik Saliev, Intel
+// spell-checker: enable
+
+== Notice
+
+Copyright (c) 2021 Intel Corporation. All rights reserved.
+
+== Status
+
+Shipping
+
+== Version
+
+Built On: {docdate} +
+Version: 1.0.0
+
+== Dependencies
+
+This feature is written against the OpenCL API Specification Version V3.0.6.
+
+This extension requires OpenCL 1.0.
+
+== Overview
+
+This extension can be used to query additional information about Intel OpenCL devices.
+The additional information may be useful to tailor a specific workload to the properties of the device.
+
+== New API Functions
+
+None.
+
+== New API Enums
+
+Possible values accepted as the _param_name_ parameter of *clGetDeviceInfo*, depending on the device type and the extension version.
+Additional queries may be added in subsequent versions of the extension:
+
+[source]
+----
+/* For GPU devices, version 1.0.0: */
+
+#define CL_DEVICE_IP_VERSION_INTEL                0x4250
+#define CL_DEVICE_ID_INTEL                        0x4251
+#define CL_DEVICE_NUM_SLICES_INTEL                0x4252
+#define CL_DEVICE_NUM_SUB_SLICES_PER_SLICE_INTEL  0x4253
+#define CL_DEVICE_NUM_EUS_PER_SUB_SLICE_INTEL     0x4254
+#define CL_DEVICE_NUM_THREADS_PER_EU_INTEL        0x4255
+#define CL_DEVICE_FEATURE_CAPABILITIES_INTEL      0x4256
+----
+
+Bitfield type describing the feature capabilities of a device.
+Additional feature flags may be added in subsequent versions of the extension.
+
+[source]
+----
+typedef cl_bitfield         cl_device_feature_capabilities_intel;
+
+/* For GPU devices, version 1.0.0: */
+
+#define CL_DEVICE_FEATURE_FLAG_DP4A_INTEL         (1 << 0)
+#define CL_DEVICE_FEATURE_FLAG_DPAS_INTEL         (1 << 1)
```

Comment:
It is not required for merging, but pointing out that there is no description of these capabilities in this document.

Reply:
Yes - we will either document these in an updated version of the spec or in a related implementation doc.  Thanks!